graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/FallbackEnvironmentArgumentResolver.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2022 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.graphql.dgs.internal
+
+import graphql.schema.DataFetchingEnvironment
+import org.springframework.core.MethodParameter
+import org.springframework.core.Ordered
+import org.springframework.core.annotation.Order
+import org.springframework.core.convert.support.DefaultConversionService
+import org.springframework.util.ClassUtils
+
+@Order(Ordered.LOWEST_PRECEDENCE)
+class FallbackEnvironmentArgumentResolver : ArgumentResolver {
```

Comment:
When is this fallback being used and why wouldn't the other resolvers not already handle this scenario?

Reply:
I will add documentation for this, but it's essentially the case where we have a parameter with no annotations. In the [existing code](https://github.com/Netflix/dgs-framework/blob/master/graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DataFetcherInvoker.kt#L96), we would check the environment to see if any argument matches the parameter name, and use that value, that is what this handler is doing. It needs to go last in the list of resolvers we try because we want to attempt conversions for `@InputArgument` first.